tests/src/Unit/SocialApiUserTest.php

```diff
@@ -0,0 +1,169 @@
+    );
+    $this->assertTrue(
+      method_exists($this->userManager, 'getDrupalUserId'),
+      'UserManager class does not implements getDrupalUserId function/method'
+    );
+  }
+
+  /**
+   * @covers Drupal\social_api\User\UserManager::setPluginId
+   */
+  public function testSetPluginId() {
+    $this->assertEquals(NULL, $this->userManager->getPluginId());
+    $this->userManager->setPluginId('social_auth_mixer');
+    $this->assertEquals('social_auth_mixer', $this->userManager->getPluginId());
+  }
+
+  /**
+   * @covers Drupal\social_api\User\UserManager::getPluginId
+   */
+  public function testGetPluginId() {
+    $this->userManager->setPluginId('social_auth_mixer');
+    $this->assertNotEquals('social_auth_reddit', $this->userManager->getPluginId());
+  }
+
+  /**
+   * Tests for class UserAuthenticator.
+   */
+  public function testUserAuthenticator() {
+    $this->assertTrue(
+      method_exists($this->userAuthenticator, 'getPluginId'),
+      'UserAuthenticator class does not implements getPluginId function/method'
+    );
+    $this->assertTrue(
+      method_exists($this->userAuthenticator, 'setPluginId'),
+      'UserAuthenticator class does not implements setPluginId function/method'
+    );
+    $this->assertTrue(
+      method_exists($this->userAuthenticator, 'getSessionKeys'),
+      'UserAuthenticator class does not implements getSessionKeys function/method'
+    );
+    $this->assertTrue(
+      method_exists($this->userAuthenticator, 'setSessionKeysToNullify'),
+      'UserAuthenticator class does not implements setSessionKeysToNullify function/method'
+    );
+    $this->assertTrue(
+      method_exists($this->userAuthenticator, 'nullifySessionKeys'),
+      'UserAuthenticator class does not implements nullifySessionKeys function/method'
+    );
+  }
+
+  /**
+   * @covers Drupal\social_api\User\UserAuthenticator::setPluginId
+   */
+  public function testSetPluginIdAuthenticator() {
+    $this->assertEquals(NULL, $this->userAuthenticator->getPluginId());
+    $this->userAuthenticator->setPluginId('social_auth_mixer');
+    $this->assertEquals('social_auth_mixer', $this->userAuthenticator->getPluginId());
+  }
+
+  /**
+   * @covers Drupal\social_api\User\UserAuthenticator::getPluginId
+   */
+  public function testGetPluginIdAuthenticator() {
+    $this->userAuthenticator->setPluginId('social_auth_mixer');
+    $this->assertNotEquals('social_auth_reddit', $this->userAuthenticator->getPluginId());
+  }
+
+  /**
+   * @covers Drupal\social_api\User\UserAuthenticator::getSessionKeys
+   */
+  public function testGetSessionKeys() {
+    $sample_session = ['h78323' => '78t2gq2g7q', 'pawdwadawd' => 'cbzhzxc'];
+    $another_session = ['mpau82ass' => 'lplaw2b21', 'caholaaw' => '018y23czs'];
+
+    $this->userAuthenticator->setSessionKeysToNullify($sample_session);
+    $this->assertNotEquals($another_session, $this->userAuthenticator->getSessionKeys());
```

Comment:
You're better off checking that what you set is what you get in this cases. `assertNotEquals` is not convenient since the session keys might be NULL and that's still no equal.